Generators/src/GeneratorHepMC.cxx

```diff
@@ -120,36 +121,134 @@ Bool_t GeneratorHepMC::generateEvent()
   return false;
 }
 
+/*****************************************************************/
+void GeneratorHepMC::pruneEvent(Select select)
```

Comment:
I think this is a fantastic idea. I was previously wondering if we couldn't prune Pythia8 events by using HepMC API which seems to make this task easy.

We can use this to verify correctness and also to benchmark the Pythia8 pruning function.

Reply:
Unfortunately, the HepMC pruning is O(N^2 log N), because of HepMCs use of `std::vector`, see also longer reply below.